src/Firebase/RemoteConfig/ApiClient.php

```diff
@@ -22,10 +21,12 @@ class ApiClient
 {
     private ClientInterface $client;
     private RemoteConfigApiExceptionConverter $errorHandler;
+    private string $baseUri;
 
-    public function __construct(ClientInterface $client)
+    public function __construct(string $projectId, ClientInterface $client)
```

Comment:
Is BC check okay with this because of the `@internal`?

Reply:
I thought about that, too, and I told myself "yes" because people shouldn't use internal classes directly.

There are other breaking changes, for example using `string` instead of `Stringable|string`, but it will be a major release after all, and I assume that most projects use simple strings anyway. I introduced `Stringable` back then because I wanted to move away from Value Object parameters (like `Uid`) without breaking projects.

My goal is being able to say in the release notes: "If you're not extending classes from the SDK and use the SDK as documented, this should be a drop-in replacement for the latest 6.x version". I hope that holds up in the end 😅

Reply:
Oh, and one other thing: I want to move away from runtime checks to static analysis checks, what are your thoughts about this?

Reply:
> I want to move away from runtime checks to static analysis checks, what are your thoughts about this?

100%! Runtime checks still can't even verify `string[]` or generics 😂 We would lose so much! I'm at the point where I consider my docblocks to be part of my code, and my static analysis tools "compile" all the code into PHP for runtime. Working with legacy stuff I end up with a fair amount of ignores or baselines, but my eventual goal is to have everything at PHPStan level 5 and then consider moving higher.

Reply:
That sounds good! I'm not sure if I was clear with what I said or if we mean the same thing (we probably do), but I will spell it out just in case, and for others that my follow our conversation:

At the moment I'm doing runtime checks to "protect" users from "doing it wrong", for example providing invalid arguments like empty strings. By removing the runtime checks, they aren't protected anymore except if they use PHPStan/Psalm themselves in their projects. This would reduce the lines of code drastically in the messaging and user management components and perhaps motivate people not using static analysis to do so.

Reply:
Understood! I think I am 90% there. Maybe this goes back to our other thread, but I use a lot of PHPStan-specific stuff that I do t always assume is enforced at the user level. So if I believe my annotation is general enough to show up in a modern IDE then I won't recheck it:
```php
/**
 * @param array<string> $names
 */
function length(array $names): int {
    return strlen(reset($names));
}
```
(Ignoring the fact that this should have been variadic)

But if it is something I expect the user to do yet I am only able to enforce it in PHPStan I typically still use an Assert:
```php
/**
 * @param class-string<Clock>
 */
function length(string $handler): int {
    Assert::instanceOf($handler, Clock::class);

    return (new $handler)->now();
}
```